corporate/views/upgrade.py

```diff
@@ -107,15 +107,15 @@ def upgrade_page(
     return response
 
 
-@self_hosting_management_endpoint
+@authenticated_remote_realm_management_endpoint
 @typed_endpoint
 def remote_realm_upgrade_page(
     request: HttpRequest,
+    remote_realm: RemoteRealm,
     *,
     realm_uuid: PathOnly[str],
```

Comment:
Hmm, I feel like the decorator should be consuming this parameter without a need to process it in the endpoint itself.

Reply:
This detail seems like a thing we could fix in a follow-up though.

@mateuszmandera FYI.